sklearn/linear_model/_logistic.py

```diff
@@ -277,31 +285,36 @@ def _logistic_regression_path(
         Cs = np.logspace(-4, 4, Cs)
 
     solver = _check_solver(solver, penalty, dual)
+    xp, _, device_ = get_namespace_and_device(X)
+    xp_y, _ = get_namespace(y)
 
     # Preprocessing.
     if check_input:
         X = check_array(
             X,
             accept_sparse="csr",
-            dtype=np.float64,
+            dtype=xp.float64,
```

Comment:
I wonder if we shouldn't try to allow `dtype=[xp.float64, xp.float32]` here, but maybe it's not stable for some solvers. We need to check in the issue tracker if that possibility was already discussed/evaluated in the past.

Reply:
I don't think this part of the code is called as we do the conversion in the fit method and we specifically set LBGFS to be xp.float64 or the max float dtype. I think we should probably use max supported float dtype here too but since it was not being called I decided not to change it here.